app/models/user.rb

```diff
@@ -12,13 +12,25 @@
 #  welcome_email_date_utc :datetime
 #  invite_date_utc        :datetime
 #  accept_terms_date_utc  :datetime
+#  state                  :string           not null
 #
 
 class User < ApplicationRecord
+  include AASM
+
   has_and_belongs_to_many :organisations
   has_many :providers, through: :organisations
 
   validates :email, presence: true
 
   audited
+
+  aasm column: 'state' do
+    state :new, initial: true
```

Comment:
Would it be better to name these states based on what we use them for? I think we only use them for terms and conditions?

Reply:
Discussed offline, it's going to be used for Ts and Cs as well but it's ok.